src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,164 @@
+package duke.main;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.ChangeMarkCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidDateFormatException;
+import duke.exception.InvalidParametersException;
+import duke.exception.MissingCommandException;
+import duke.exception.MissingParametersException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/** Deals with making sense of user command, manages error handling based on user input */
+public class Parser {
+    private static String[] splitText;
+
+    /**
+     * Interprets user input, returns Command class if found.
+     * The format for this parser is "[main command] [task] /[sub command] [parameters for sub command]".
+     * Order of multiple sub commands are ignored, and not case-sensitive.
+     *
+     * @param inputString Input String to be interpreted.
+     * @return Command class to run.
+     * @throws InvalidCommandException When command is invalid.
+     * @throws MissingParametersException When there are missing parameters for either main or sub commands.
+     * @throws InvalidParametersException When there are invalid parameters for sub commands.
+     * @throws InvalidDateFormatException When there are invalid date formats for sub commands.
+     * @throws MissingCommandException When sub command for main command is missing.
+     */
+    public static Command parse(String inputString) throws InvalidCommandException, MissingParametersException,
+            InvalidParametersException, InvalidDateFormatException, MissingCommandException {
+        // Split text into two.
+        // With index 0 be the first word of user input, and index 1 containing the rest of the string.
+        inputString = inputString.toLowerCase();
+        splitText = inputString.split(" ", 2);
```

Comment:
Maybe you can put this.splitText to distinguish the local and global variables since they are put next to each other this time?

Reply:
apologies, can you elaborate on this? I'm not very sure which variable is considered to be global in this case.

Reply:
sorry for the confusion, since splitText is a member of the class and inputString is a local variable to this function, maybe you can use this.splitText to specify that it is a member of the class because I was confused for a second thinking whether it is a local variable or not.